ql/src/java/org/apache/hadoop/hive/ql/ddl/table/constraint/ConstraintsUtils.java

```diff
@@ -282,6 +298,48 @@ private static List<ConstraintInfo> generateConstraintInfos(ASTNode child, List<
     return constraintInfos;
   }
 
+  static class ConstraintExpressionContext implements NodeProcessorCtx {
+    private UnparseTranslator unparseTranslator;
+
+    public ConstraintExpressionContext(UnparseTranslator unparseTranslator) {
+      this.unparseTranslator = unparseTranslator;
+    }
+
+    public UnparseTranslator getUnparseTranslator() {
+      return unparseTranslator;
+    }
+  }
+
+  private static UnparseTranslator collectUnescapeIdentifierTranslations(ASTNode node)
```

Comment:
should we mark inner static classes as final as well

Reply:
It may makes sense if the class is exposed via an API but `ConstraintsUtils` is for internal usage only.